### ISSUE TITLE
Missing MTU response

### ISSUE THREAD

**Issue by Stretchdude** [edrosten/libblepp#31]
Hello,

I found (after upgrading our BLE device to newer BLE-Stack), that a request for MTU exchange is treated as an error. The state machine is bailing out with "unexpected response". After some debugging I manage to hold the connection alive by sending a faked response (which sends static 24 as MTU, the "old value" from the device). 

- Is there a cleaner way to handle the MTU request? Bluez-Gatttool responses with "request not supported" but I wasn't able to produce this in libblepp.
- Why  is the request killing the state machine? Is the timming just bad luck or is the request wrong in this time slot?

I append my hack as patch to let you have a look.

Regards
Ben

[liblepp_mtu_hack.patch.txt](https://github.com/edrosten/libblepp/files/2159247/liblepp_mtu_hack.patch.txt)

trace of the hack:
![mtu_request_ws_trace](https://user-images.githubusercontent.com/35837112/42220671-ede7fede-7ecf-11e8-8488-72ac6787837d.png)

**Comment by Stretchdude** [comment]
Hello,

thanks for your reply. I had make kind of progress on this issue. But still it can get cought in a deadlock if there is unfortunate timing.

**Comment by Stretchdude** [comment]
Sure, appended.
Please tell me if I can try something. This is made with HEAD from your github. I could also provide a Wireshark trace from my bluetooth0 device if that is of any help (but no real BLE sniffer available). 
[liblepp_trace_mtu_request_fail.txt](https://github.com/edrosten/libblepp/files/2184318/liblepp_trace_mtu_request_fail.txt)

**Comment by Stretchdude** [comment]
Hi, updates :-)
I managed to find out, that libblepp is not even receiving some of the MTU requests. I think they may get thrown away by the kernel because connection is not ready to receive but the device is very quick with sending them. 
A sleep on the device side helped with the problem, but who wants to have a sleep as a fix?
I don't know how to fix this without diggin into bluez kernel stuff. (and not diggin into bluez was the reason for me to search for libblepp ;-)
I workarounded it now with sending just a denial always after connect. Also there is the change which handles MTU request in the state machine still active.
Ugly but works for me. I guess this is a hack (again), so I don't think you like to have a pull request.
Nevertheless I'll append a patch and push it to my own fork, after cleaning. If somebody has the same issue...
